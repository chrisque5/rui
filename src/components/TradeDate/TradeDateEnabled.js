import PropTypes from 'prop-types';
import { InfoCircleFilled } from '@ant-design/icons';
import {
  Checkbox, Form, Tooltip,
} from 'antd';

const TradeDateEnabled = ({
  id, onChange,
}) => (
  <Form.Item
    className="trade-date-enabled"
    valuePropName="checked"
    name={id}
    initialValue={false}
  >
    <Checkbox onChange={onChange} data-testid={`chk-${id}`}>
      <span className="label-text">Override Trade Date</span>
      <Tooltip
        title="Override Trade Date/Time"
        mouseEnterDelay={0.3}
        overlayStyle={{ fontSize: '0.8em', padding: 0, margin: 0 }}
      >
        <InfoCircleFilled />
      </Tooltip>

    </Checkbox>
  </Form.Item>

);

TradeDateEnabled.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TradeDateEnabled;
