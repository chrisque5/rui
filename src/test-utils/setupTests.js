// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
