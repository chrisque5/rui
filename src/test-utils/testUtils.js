/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
