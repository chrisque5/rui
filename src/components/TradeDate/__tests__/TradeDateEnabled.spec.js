import { renderWithForm, fireEvent, waitFor } from 'test-utils/testUtils';
import TradeDateEnabled from '../TradeDateEnabled';

const mockForm = {};
const defaultProps = { form: mockForm, id: 'tradeDateEnabled', onChange: () => jest.fn() };

test('<TradeDateEnabled /> renders snapshot as expected', () => {
  const { container } = renderWithForm(<TradeDateEnabled {...defaultProps} />);
  expect(container).toMatchSnapshot();
});

test('<TradeDateEnabled /> handles clicking of checkbox', async () => {
  const { getByLabelText } = renderWithForm(<TradeDateEnabled {...defaultProps} />);
  const checkbox = await waitFor(() => getByLabelText('Override Trade Date'));
  expect(checkbox.checked).toEqual(false);

  fireEvent.click(checkbox);
  expect(checkbox.checked).toEqual(true);

  fireEvent.click(checkbox);
  expect(checkbox.checked).toEqual(false);
});
