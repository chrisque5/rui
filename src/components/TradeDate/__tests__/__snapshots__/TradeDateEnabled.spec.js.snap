// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeDateEnabled /> renders snapshot as expected 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item trade-date-enabled"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <label
              class="ant-checkbox-wrapper"
            >
              <span
                class="ant-checkbox"
              >
                <input
                  class="ant-checkbox-input"
                  data-testid="chk-tradeDateEnabled"
                  id="tradeDateEnabled"
                  type="checkbox"
                  value=""
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span>
                <span
                  class="label-text"
                >
                  Override Trade Date
                </span>
                <span
                  aria-label="info-circle"
                  class="anticon anticon-info-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="info-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                    />
                  </svg>
                </span>
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
